users/views.py

```diff
@@ -1,39 +1,32 @@
 from django_filters.rest_framework import DjangoFilterBackend
 from rest_framework import filters
-from rest_framework import status
 from rest_framework.generics import CreateAPIView, ListAPIView, UpdateAPIView, DestroyAPIView, RetrieveAPIView
 from rest_framework.permissions import AllowAny
-from rest_framework.response import Response
 
-from materials.models import Course, Lesson
 # from config.settings import EMAIL_HOST_USER
 from users.models import Payments
 from users.models import User
 from users.serializer import PaymentsSerializer, UserSerializer
+from users.services import create_stripe_price, stripe_create_product
 
 
 class PaymentsCreateApiView(CreateAPIView):
     queryset = Payments.objects.all()
     serializer_class = PaymentsSerializer
 
-    def create(self, request, *args, **kwargs):
-        course_id = request.data.get('course_id')
-        lesson_id = request.data.get('lesson_id')
-        payment_amount = request.data.get('payment_amount')
-        payment_method = request.data.get('payment_method', 'transfer')  # Устанавливаем значение по умолчанию
-
-        if course_id:
-            course = Course.objects.get(id=course_id)
-            payment = Payments.objects.create(user=request.user, course=course, payment_amount=payment_amount,
-                                              payment_method=payment_method)
-        elif lesson_id:
-            lesson = Lesson.objects.get(id=lesson_id)
-            payment = Payments.objects.create(user=request.user, lesson=lesson, payment_amount=payment_amount,
-                                              payment_method=payment_method)
-        else:
-            return Response({'error': 'Необходимо указать курс или урок'}, status=status.HTTP_400_BAD_REQUEST)
-
-        return Response(PaymentsSerializer(payment).data, status=status.HTTP_201_CREATED)
+    def perform_create(self, serializer):
+        payment = serializer.save(user=self.request.user)
+        print(payment.course, payment.lesson)
+        if payment.course is not None:
+            product = stripe_create_product(payment.course.title)
+        elif payment.lesson is not None:
+            product = stripe_create_product(payment.lesson.title)
+
+        price = create_stripe_price(product, payment.payment_amount)
+        session_id, payment_link = create_stripe_price(price)
+        payment.session_id = session_id
+        payment.link = payment_link
+        payment.save()
```

Comment:
Так немного оптимальнее. Мы обновляем только два поля
```suggestion
        payment.save(updated_fields=['session_id', 'link')
```